core/modules/wdb/QueryRunner.cc

```diff
@@ -233,8 +233,15 @@ bool QueryRunner::_fillRows(MYSQL_RES* result, int numFields, uint& rowCount, si
         tSize += rawRow->ByteSize();
         ++rowCount;
 
+        unsigned int szLimit = proto::ProtoHeaderWrap::PROTOBUFFER_DESIRED_LIMIT;
+
+        // Use small blocks until it is considered a large result.
+        if (!_largeResult) {
+            szLimit = std::min(szLimit, _initialBlockSize);
+        }
+
         // Each element needs to be mysql-sanitized
-        if (tSize > proto::ProtoHeaderWrap::PROTOBUFFER_DESIRED_LIMIT) {
+        if (tSize > szLimit) {
```

Comment:
How is szLimit set? Is it always guaranteed to be less than the protobuffer hard limit?